### ISSUE TITLE
Feature Request - Label K-fold cross validation

### ISSUE THREAD

**Issue by aquintero** [topepo/caret#465]
Essentially this from the scikit-learn python library: http://scikit-learn.org/stable/modules/cross_validation.html#label-k-fold

**Comment by topepo** [comment]
I'm working on this now. It would be good for you to send a small test case (with expected results) that I can use to test with.

**Comment by aquintero** [comment]
Thanks for looking into it. Here is a simple example with a possible result. Note that there could be several possible fold separations that work. You just want to guarantee that data points with same label appear in same fold (i.e. not separated into different folds).

[example.zip](https://github.com/topepo/caret/files/412159/example.zip)

Some edge cases you might want to think about are when the number of folds is larger than the number of labels, and when the distribution of labels is skewed (like half of the data set is one label). I'm not sure about the best way to approach these cases. Maybe there needs to be stratification involved? [I would look at how sci-kit does it.](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/model_selection/_split.py#L421)

**Comment by topepo** [comment]
I believe that this was solved in issue #540 but please verify
